### ISSUE TITLE
Method invoke context, need better way to set to Meteor.userId, for hooks

### ISSUE THREAD

**Issue by zeroasterisk** [stubailo/meteor-rest#119]
We are currently setting a "fake" invoke context here:
https://github.com/stubailo/meteor-rest/blob/devel/packages/rest/rest.js#L199-L222

This works well enough for the Method call, but not for anything subsequent... for example, when using Meteor Collection Hooks, it is not able to determine the `userId`
https://github.com/matb33/meteor-collection-hooks/blob/master/collection-hooks.js#L28-L51

I tried to wrap `addHTTPMethod` and it's inner handler function in `Meteor.bindEnvironment` but since it's a fake context anyway, I don't think it did anything for me.

I tried to set `Meteor.userId = userId` but that didn't seem to help much either (and kinda freaked me out).

---

Is there some means of either setting up a better and more rhobust Meteor context in the REST API, or perhaps telling MeteorCollectionHooks about how to find the `userId` set into the method's context, but not available to the collection's context.

?

**Comment by zeroasterisk** [comment]
Here's a fully demonstrating test repo:
https://github.com/zeroasterisk/meteor-rest-and-collection-hooks

Just clone and start meteor, it creates a user, a doc, and uses HTTP to login, and then attempt the method...

It shows that method works fine, and is authenticated, but the collection hook does not.

thoughts?

crosspost to https://github.com/matb33/meteor-collection-hooks/issues/178

**Comment by hluz** [comment]
@zeroasterisk did you (or anyone else listening) found any workable solution for this issue?

**Comment by zeroasterisk** [comment]
No not really.

**Comment by hluz** [comment]
Maybe this will fix it: https://github.com/stubailo/meteor-rest/pull/118

**Comment by dpatte** [comment]
yes, there is a pull request 118 that fixes the same problem in simple:rest, and which works for me.
